Makefile

```diff
@@ -0,0 +1,29 @@
+# Makefile
+THIS_FILE := $(lastword $(MAKEFILE_LIST))
+include common.mk
+SHELL := /bin/bash
+
+RUNTEST=python -m unittest -v -b
+
+ALLMODULES=$(patsubst %.py, %, $(wildcard test_*.py))
+
+all:
+	@echo "running tests..."
+	${RUNTEST} ${ALLMODULES}
+
+% : test_%.py
```

Comment:
Does this do anything? I don't see it being invoked, although I may be missing something.

Reply:
I found this with `all` and renamed it to either `test` or tests` and it stopped working, so I put `all` back in. It seems a convention (see https://www.gnu.org/software/make/manual/html_node/Standard-Targets.html#Standard-Targets). However, in this Makefile I wouldn't consider `all` to compile the entire program (doesn't compile anything actually). I find out which one of the two test/tests works and try to replace it.

Reply:
If all is the convention, I don't think it should be on the tests.

I think all should do everything short of the deploy -- install the virtualenv and run the tests.

If someone clones the repo from scratch and just does a `make`, won't it fail? Because that will just run the tests, but will have not set up the virtualenv...

I agree on the need for `all`, especially after reading the link you provided, but I don't think it's functionality should be to run the tests.